pyscf/pbc/df/rsdf_helper.py

```diff
@@ -0,0 +1,1297 @@
+        nbas = mol.nbas//2
+        n2 = nbas*(nbas+1)//2
+        if len(dijs_lst) != n2:
+            raise RuntimeError("dijs_lst has wrong len (expecting %d; got %d)" % (n2, len(dijs_lst)))
+        intor = "int2e"
+        Qs = [None] * n2
+        ij = 0
+        for i in range(nbas):
+            for j in range(i+1):
+                j_ = j + nbas
+                shls_slice = (i,i+1,j_,j_+1,i,i+1,j_,j_+1)
+                dijs = dijs_lst[ij]
+                Qs[ij] = [compute1_(mol, dij, intor, shls_slice, omega, safe)
+                          for dij in dijs]
+                ij += 1
+
+    return Qs
+def get_schwartz_dcut(bas_lst, cellvol, omega, precision, r0=None, safe=True,
+                      vol_correct=False):
+    """ Given a list of basis, determine cutoff radius for the Schwartz Q
+    between each unique shell pair to drop below "precision". The Schwartz
+    Q is define:
+        Q = 2-norm[ (ab|ab) ]^(1/2)
+
+    Return:
+        1d array of length nbas*(nbas+1)//2 with nbas=len(bas_lst).
+    """
+    mol = mol_gto.M(atom="H 0 0 0; H 0 0 0", basis=bas_lst)
+    nbas = len(bas_lst)
+    n2 = nbas*(nbas+1)//2
+
+    es = np.array([mol.bas_exp(i).min() for i in range(nbas)])
+    etas = 1/(1/es[:,None] + 1/es)
+# >>>>>>> debug block
+    if vol_correct:
+        fac = 2*np.pi/cellvol
+    else:
+        fac = 1.
+# <<<<<<<
+
+    intor = "int2e"
+    def estimate1(ish,jsh,R0,R1):
+        shls_slice = (ish,ish+1,nbas+jsh,nbas+jsh+1,
+                      ish,ish+1,nbas+jsh,nbas+jsh+1)
+        prec0 = precision * min(etas[ish,jsh],1.)
+        def fcheck(R):
+            mol._env[mol._atm[1,mol_gto.PTR_COORD]] = R
+            I = get_norm(fintor_sreri(mol, intor, shls_slice, omega, safe))**0.5
+# >>>>>>> debug block
+            I *= fac
+# <<<<<<<
+            prec = prec0 * min(1./R,1.)
+            return I < prec
+        return binary_search(R0, R1, 1, True, fcheck)
+
+    if r0 is None: r0 = 30
+    R0 = r0 * 0.3
+    R1 = r0
+    dcuts = np.zeros(n2)
+    ij = 0
+    for i in range(nbas):
+        for j in range(i+1):
+            dcuts[ij] = estimate1(i,j,R0, R1)
+            ij += 1
+    return dcuts
+def make_dijs_lst(dcuts, dstep):
+    return [np.arange(0,dcut,dstep) for dcut in dcuts]
+def get_bincoeff(d,e1,e2,l1,l2):
+    d1 = -e2/(e1+e2) * d
+    d2 = e1/(e1+e2) * d
+    lmax = l1+l2
+    cbins = np.zeros(lmax+1)
+    for l in range(0,lmax+1):
+        cl = 0.
+        lpmin = max(-l,l-2*l2)
+        lpmax = min(l,2*l1-l)
+        for lp in range(lpmin,lpmax+1,2):
+            l1p = (l+lp) // 2
+            l2p = (l-lp) // 2
+            cl += d1**(l1-l1p)*d2**(l2-l2p) * comb(l1,l1p) * comb(l2,l2p)
+        cbins[l] = cl
+    return cbins
+def get_3c2e_Rcuts_for_d(mol, auxmol, ish, jsh, dij, cellvol, omega, precision,
+                         fac_type, Qij, Rprec=1,
+                         vol_correct=False, eta_correct=True, R_correct=True):
```

Comment:
What are fac_type, Qij, Rprec, vol_correct, eta_correct, R_correct? Please add docs for these arguments.

Reply:
fixed in [this commit](https://github.com/pyscf/pyscf/pull/1034/commits/94585b16b1987beda31e273b0e5e2bc003e76f01)